orca-core/orca-core.gradle

```diff
@@ -54,6 +54,8 @@ dependencies {
   implementation("com.jayway.jsonpath:json-path:2.2.0")
   implementation("org.yaml:snakeyaml")
   implementation("org.codehaus.groovy:groovy")
+  implementation("net.javacrumbs.shedlock:shedlock-spring:2.2.0")
```

Comment:
Could you please provide an overview of this dependency, and why it's a good choice, other alternatives, etc.

Reply:
The alternative I know for distributed services locks:
* Raft consensus protocols
* jgroups
* Quartz (which is ... full CRON scheduler)
* This lib :) which is pure SQL it seems

I don't know that we really want raft (aka https://ratis.apache.org/) or jgroups (which.. keycloak uses.. and is kinda a geast).  Quartz is a bit overkill.  I'd be open to another distributed lock lib - ideally though this is one that replaces any OTHER distributed lock needs LIKE Igor's current system for polling locks.

Reply:
Hey there!

@jasonmcintosh 
Thanks for sharing alternative options! All of them would get the job done!

I was looking for a simple library that focuses on providing a locking mechanism without any additional scheduling features, which I didn't need. After some research, I found a library that perfectly fits my needs: it is lightweight and easy to use, and it provides a reliable locking mechanism that can be integrated with Redis (maybe it can replace LockingManager?) and other storage options. This library has the potential to become the foundation of a new implementation that could replace the current messy locking situation. In the future, by centralizing the locking options within kork, we can achieve better code organization and reduce the risk of errors. Currently, the locking options are spread across the kork and apps repositories, which makes it hard to manage and maintain them. By using this library, we can simplify the locking process.

If you have any questions or concerns, please don't hesitate to reach out. I am happy to help or explain more!

Reply:
ONLY other question is the 2.2.0 version.  Which I'd expect to match spring maybe?  Which seems like it's not quite matching?  At that point, it might need a kork change :(

Reply:
Bumped the version to 4.44. Cannot bump it to 5.2, because it requires Java 17.